cypress/integration/purchase/purchase_order_complete_change_complete.js

```diff
@@ -0,0 +1,170 @@
+    cy.fixture('product/packing_material.json').then(packingMaterialJson => {
+      Object.assign(new PackingMaterial(), packingMaterialJson)
+        .setName(packingMaterialName)
+        .setProduct(productForPackingMaterial)
+        .apply();
+    });
+    cy.fixture('product/packing_instructions.json').then(packingInstructionsJson => {
+      Object.assign(new PackingInstructions(), packingInstructionsJson)
+        .setName(packingInstructionsName)
+        .apply();
+    });
+    cy.fixture('product/packing_instructions_version.json').then(pivJson => {
+      Object.assign(new PackingInstructionsVersion(), pivJson)
+        .setName(packingInstructionsName)
+        .setPackingInstructions(packingInstructionsName)
+        .setPackingMaterial(packingMaterialName)
+        .apply();
+    });
+    cy.fixture('product/simple_productCategory.json').then(productCategoryJson => {
+      Object.assign(new ProductCategory(), productCategoryJson)
+        .setName(productCategoryName)
+        .setValue(productCategoryValue)
+        .apply();
+    });
+
+    Builder.createBasicProductEntitiesWithCUTUAllocation(
+      productCategoryName,
+      productCategoryValue,
+      priceListName,
+      productName1,
+      productValue1,
+      productType,
+      packingInstructionsName
+    );
+    cy.fixture('sales/simple_vendor.json').then(vendorJson => {
+      new BPartner({ name: vendorName })
+        .setVendor(true)
+        .setVendorPricingSystem(priceSystemName)
+        .setVendorDiscountSchema(discountSchemaName)
+        .setPaymentTerm('30 days net')
+        .addLocation(new BPartnerLocation('Address1').setCity('Cologne').setCountry('Deutschland'))
+        .apply();
+    });
+    cy.readAllNotifications();
+  });
+  it('Create a purchase order and complete it', function() {
+    cy.visitWindow('181', 'NEW');
+    cy.get('#lookup_C_BPartner_ID input')
+      .type(vendorName)
+      .type('\n');
+    cy.contains('.input-dropdown-list-option', vendorName).click();
+
+    cy.selectInListField('M_PricingSystem_ID', priceSystemName, false, null, true);
+    const addNewText = Cypress.messages.window.batchEntry.caption;
+    cy.get('.tabs-wrapper .form-flex-align .btn')
+      .contains(addNewText)
+      .should('exist')
+      .click();
+    cy.get('.quick-input-container .form-group').should('exist');
+    cy.writeIntoLookupListField('M_Product_ID', productName1, productName1, false, false, null, true);
+
+    cy.get('.form-field-Qty')
+      .click()
+      .find('.input-body-container.focused')
+      .should('exist')
+      .find('i')
+      .eq(0)
+      .click();
+    cy.server();
+    cy.route('POST', `/rest/api/window/${purchaseOrders.windowId}/*/${purchaseOrders.orderLineTabId}/quickInput`).as(
+      'resetQuickInputFields'
+    );
+    cy.get('.form-field-Qty')
+      .find('input')
+      .should('have.value', '0.1')
+      .clear()
+      .type('5{enter}');
+    cy.wait(8000);
```

Comment:
instead of waiting a static 8 seconds, can't we wait for for the number of order line rows to be one (and not zero anymore)?
Or maybe for something else, like incoming order lines data from the API?

Reply:
The only solution I have found was this: cy.get('#lookup_M_Product_ID .input-dropdown').should('not.have.class', 'input-block'); - this way the wait is removed. I have replaced it in all the tests in which was used after entering a quantity.